HIP/hip-679.md

```diff
@@ -0,0 +1,215 @@
+    
+Example:
+    accountBalances/balance0.0.3/
+    eventsStreams/events_0.0.3/
+    recordstreams/record0.0.3/
+```
+    
+Proposed Structure:
+```
+    {network}/{shard}/{nodeID}/balance/
+    {network}/{shard}/{nodeID}/event/
+    {network}/{shard}/{nodeID}/record/
+    
+Example:
+
+    Standard network:
+    mainnet/0/3/balance/
+    mainnet/0/3/event/
+    mainnet/0/3/record/
+    
+    Resettable network:
+    testnet-2023-01/0/3/balance/
+    testnet-2023-01/0/3/event/
+    testnet-2023-01/0/3/record/    
+```
+##### Problems with the current structure naming:
+- Stream name is duplicated in `eventsStreams` and `recordstreams`. We are missing stream from `accountBalances`.
+- Casing is inconsistent in `eventsStreams` and `recordstreams`.
+- After the separator we are repeating stream names. `events_0.0.3` has an underscore where as the others don't.
+
+The new structure proposes consistent naming. In the future when we start supporting multiple shards, the new bucket structure will automatically work for it without any required changes.
+
+##### Proposed naming structure:
+- The network name in the bucket path will not change for non-resettable networks. 
+- For networks that can be reset the network key in the path will start wth the network name prefix followed by a timestamp, as stated in the above testnet example. On a reset, to get the latest network one would need to filter by the network prefix (e.g. `testnet`) then sort the output to get the latest network to read from.
+- The granularity of the timestamp would depend on how often that particular network is reset.
+- The remaining bucket path after the network name would be similar for all networks.
+- Using the network name would be strictly internal to mirror node.
+
+It is thus proposed to be a phased approach with three phases.
+
+#### Phase One:
+Update mirror node downloader to read from the new `{network}/{shard}/{nodeID}/*` nodeId based path as well as support reading from `*/*0.0.3/` account id based path.
+
+##### Algorithm:
+ - Add a `pathType` property with values `AUTO`, `ACCOUNT_ID`, and `NODE_ID` with a default of `ACCOUNT_ID`
+ - Later once consensus nodes get closer to writing to the new path adjust default to `AUTO`
+ - Add a `pathRefreshInterval` property with a default value of `1m`
+ - If pathType is `AUTO`:
+    - Attempt to list signatures from  a path with node account IDs
+    - If successful, update a per node timestamp and boolean to ensure account ID is used until pathRefreshInterval elapses
+    - If no results, attempt to list signatures from a path with node IDs
+    - If successful, set `pathType` as `NODE_ID` so the node ID path is used going forward
+   
+In the existing scenario the default pathType would be `ACCOUNT_ID`, which would then be changed to AUTO when we are close to approaching Phase two.
+
+#### Phase Two:
+Update the consensus nodes to write stream files to the new `{network}/{shard}/{nodeID}/*` nodeId based path.
```

Comment:
I believe this may require updates to mirror.py, as I don't believe it currently recurses directories. it may be possible to account for in the deployment and target mirror.py to watch specifically the `{nodeID}` folder, provided there are no further sub-directories after this point

Reply:
I don't think there's any need for recursion. The watch folder name and logic can stay exactly as is. It's only the S3 destination name that is changing, which should hopefully just be a config tweak.